### PR TITLE
Use question helper for passwords in commands

### PR DESCRIPTION
**Pull request by mynameisbogdan**
### 1. Why is this change necessary?
To fix max attempts functionality in `user:create` and `user:change-password`, and to ask interactively for a password `store:login`. 

### 2. What does this change do, exactly?
Adds a password question to `store:login`, and password validation to `user:create` and `user:change-password` in order to fix max attempts functionality 

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Framework/Store/Command/StoreLoginCommand.php [hunk 1]
```diff
@@ -74,6 +70,21 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+                }
+
+                return $value;
+            });
+            $passwordQuestion->setHidden(true);
+            $passwordQuestion->setMaxAttempts(3);
+
+            $password = $io->askQuestion($passwordQuestion);
+        }
+
```
Comment (NiklasLimberg) [inline src/Core/Framework/Store/Command/StoreLoginCommand.php:87]: You don't need to check this, because the password is requierd. So please delete these lines

File: src/Core/System/User/Command/UserChangePasswordCommand.php [hunk 2]
```diff
@@ -44,21 +43,28 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+            $passwordQuestion = new Question('Enter new password for user');
+            $passwordQuestion->setValidator(static function ($value): string {
+                if ($value === null || trim($value) === '') {
+                    throw new \RuntimeException('The password cannot be empty');
+                }
 
-        if (empty($password)) {
-            $passwordQuestion = new Question('New password for the user');
+                return $value;
+            });
```
Comment (NiklasLimberg) [inline src/Core/System/User/Command/UserChangePasswordCommand.php:54]: this too cannot be empty, because it is requiered

File: src/Core/System/User/Command/UserCreateCommand.php [hunk 3]
```diff
@@ -51,8 +49,15 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
-            $passwordQuestion = new Question('Password for the user');
+        if (!$password) {
+            $passwordQuestion = new Question('Enter password for user');
+            $passwordQuestion->setValidator(static function ($value): string {
+                if ($value === null || trim($value) === '') {
+                    throw new \RuntimeException('The password cannot be empty');
+                }
+
+                return $value;
+            });
```
Comment (NiklasLimberg) [inline src/Core/System/User/Command/UserCreateCommand.php:60]: Same here

**Comment by NiklasLimberg** [comment]
Hi,
@mynameisbogdan I imported your commit and @shyim said those lines are unnecessary

File: changelog/_unreleased/2021-09-04-use-question-helper-for-passwords-in-commands.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+title: Use question helper for passwords in commands
+author: mynameisbogdan
+author_email: mynameisbogdan@protonmail.com
```
Comment (NiklasLimberg) [inline changelog/_unreleased/2021-09-04-use-question-helper-for-passwords-in-commands.md:5]: Oh and it would be nice if you could add 'issue: NEXT-17106``  after the title line

File: src/Core/Framework/Store/Command/StoreLoginCommand.php (hunk 1 above)
Comment (mynameisbogdan) [reply src/Core/Framework/Store/Command/StoreLoginCommand.php:87]: Hello,

Please keep in mind that even if the option `password` has the `required` flag you can still ignore passing it as an argument and therefor it will be `null`. 

The whole purpose of this PR is to add a interactive option to supply a password for those who would like to keep sensitive passwords from leaking in the terminal history. 

Meaning your request to remove these lines invalidates this purpose.

File: src/Core/System/User/Command/UserChangePasswordCommand.php (hunk 2 above)
Comment (mynameisbogdan) [reply src/Core/System/User/Command/UserChangePasswordCommand.php:54]: Please take a look here https://streamable.com/256gh0

Without this check for an empty password, `setMaxAttempts` is not doing it's intended purpose.

File: src/Core/System/User/Command/UserCreateCommand.php (hunk 3 above)
Comment (mynameisbogdan) [reply src/Core/System/User/Command/UserCreateCommand.php:60]: Please take a look here https://streamable.com/256gh0

Without this check for an empty password, `setMaxAttempts` is not doing it's intended purpose.

File: src/Core/Framework/Store/Command/StoreLoginCommand.php (hunk 1 above)
Comment (shyim) [reply src/Core/Framework/Store/Command/StoreLoginCommand.php:87]: Hey,

that's ok. Can you add a test for the password empty case?

**Comment by NiklasLimberg** [comment]
Well,
I'm sorry I just blindly trusted @shyim. As shyim said you need to provide tests for the missing password cases, so the incomplete still applies

File: src/Core/Framework/Store/Command/StoreLoginCommand.php (hunk 1 above)
Comment (mynameisbogdan) [reply src/Core/Framework/Store/Command/StoreLoginCommand.php:87]: Hey @shyim,

I just pushed tests for all three commands.

**Comment by mynameisbogdan** [comment]
Hey @NiklasLimberg,

No worries, at first I was under the same impression that being required should throw some error. I wanted to make it optional just to implement the interactive asking of the password, but I found out it worked just okay even if with required flag. 
Even now I had to test again inside of a docker container and on barebone just to be sure isn't a issue with my setup. 

Also `setMaxAttempts` I remember to work for me in the past without a value check. I suspect something changed in the recent `symfony/console` releases.

File: src/Core/Framework/Store/Command/StoreLoginCommand.php [hunk 5]
```diff
@@ -74,6 +70,21 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
         $user = $input->getOption('user');
 
+        if (!$password) {
+            $passwordQuestion = new Question('Enter password');
+            $passwordQuestion->setValidator(static function ($value): string {
+                if ($value === null || trim($value) === '') {
+                    throw new \RuntimeException('The password cannot be empty');
+                }
+
+                return $value;
```
Comment (Dominik28111) [inline src/Core/Framework/Store/Command/StoreLoginCommand.php:80]: these lines are not covered

File: src/Core/System/User/Command/UserChangePasswordCommand.php [hunk 6]
```diff
@@ -44,21 +43,28 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+        if (!$password) {
+            $passwordQuestion = new Question('Enter new password for user');
+            $passwordQuestion->setValidator(static function ($value): string {
+                if ($value === null || trim($value) === '') {
+                    throw new \RuntimeException('The password cannot be empty');
+                }
 
-        if (empty($password)) {
-            $passwordQuestion = new Question('New password for the user');
+                return $value;
```
Comment (Dominik28111) [inline src/Core/System/User/Command/UserChangePasswordCommand.php:53]: these lines are not covered

File: src/Core/System/User/Command/UserCreateCommand.php [hunk 7]
```diff
@@ -51,8 +49,15 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
-        if (empty($password)) {
-            $passwordQuestion = new Question('Password for the user');
+        if (!$password) {
+            $passwordQuestion = new Question('Enter password for user');
+            $passwordQuestion->setValidator(static function ($value): string {
+                if ($value === null || trim($value) === '') {
+                    throw new \RuntimeException('The password cannot be empty');
+                }
+
+                return $value;
```
Comment (Dominik28111) [inline src/Core/System/User/Command/UserCreateCommand.php:59]: these lines are not covered

**Review by Dominik28111 — CHANGES_REQUESTED** [review]

File: src/Core/Framework/Store/Command/StoreLoginCommand.php (hunk 5 above)
Comment (mynameisbogdan) [reply src/Core/Framework/Store/Command/StoreLoginCommand.php:80]: Fixed.

File: src/Core/System/User/Command/UserCreateCommand.php (hunk 7 above)
Comment (mynameisbogdan) [reply src/Core/System/User/Command/UserCreateCommand.php:59]: Fixed.

File: src/Core/System/User/Command/UserChangePasswordCommand.php (hunk 6 above)
Comment (mynameisbogdan) [reply src/Core/System/User/Command/UserChangePasswordCommand.php:53]: Fixed.

**Comment by NiklasLimberg** [comment]
Hi @mynameisbogdan,
i forgot to change the label. Thanks for your contribution
